public/app/features/alerting/unified/utils/rule-id.ts

```diff
@@ -104,7 +119,7 @@ export function parse(value: string, decodeFromUri = false): RuleIdentifier {
   }
 
   if (parts.length === 5) {
-    const [prefix, ruleSourceName, namespace, groupName, hash] = parts.map(unesacapeDollars);
+    const [prefix, ruleSourceName, namespace, groupName, hash] = parts.map(unescapeDollars).map(unescapePathSeparators);
```

Comment:
can we add a test for this or is it very tricky? at least to make sure the path separator characters are always unescaped.

Reply:
Yep, I've added a few tests in [rule-id.test.ts](https://github.com/grafana/grafana/pull/58141/files#diff-2b7b34d539bea0197a94f3a85e05a7bcf70de3cbf838bbbb45280a759fd5b242) – I can certainly add some more, which ones do you think are missing?